cfgrammar/src/lib/yacc/grammar.rs

```diff
@@ -557,6 +557,28 @@ where
         }
     }
 
+    /// Returns the string representation of a given production `pidx`.
+    pub fn pp_prod(&self, pidx: PIdx<StorageT>) -> String {
+        let mut sprod = String::new();
+        let ridx = self.prod_to_rule(pidx);
+        sprod.push_str(self.rule_name(ridx));
+        sprod.push_str(":");
+        for sym in self.prod(pidx) {
+            let s = match sym {
+                Symbol::Token(tidx) => {
+                    if *tidx == self.eof_token_idx() {
```

Comment:
The `if` statement is unnecessary: just `self.token_name(*tidx).unwrap()` is needed (it'll `panic` if it hits `eof_token_idx`).